src/javascript-extract.js

```diff
@@ -33,20 +34,42 @@ function extractConcatenatedStrings(value, allTokens, index) {
 }
 
 
-function getGettextEntriesFromScript(script) {
+function getGettextEntriesFromScript(script, parser = 'auto') {
+
   const allTokens = [];
 
-  const ACORN_OPTIONS = {
-    ecmaVersion: 10,
-    sourceType: 'module',
-    locations: true,
-    onToken: allTokens,
-    plugins: {
-      stage3: true,
-    },
-  };
+  switch (parser) {
+  case 'auto':
+    try {
+      return getGettextEntriesFromScript(script, 'acorn');
+    } catch (e) {
+      return getGettextEntriesFromScript(script, 'babel');
+    }
```

Comment:
I'd very much rather have 2 sub-functions, one that parses with babel, another that parses with acorn, and that you call them instead of having the function calling itself with different arguments.

Also, in the `switch` statemnt you could just leave this calling of the two parsers as default instead of using the 'auto' keyword :)